contrib/mpi-proxy-split/two-phase-algo.cpp

```diff
@@ -309,16 +311,27 @@ TwoPhaseAlgo::stop(MPI_Comm comm)
 }
 
 phase_t
-TwoPhaseAlgo::waitForNewStateAfter(phase_t oldState)
+TwoPhaseAlgo::waitForNewStateAfter(phase_t oldState, int timeout_ms)
 {
   lock_t lock(_phaseMutex);
-  // The user thread will notify us if transition to any of these states
-  _phaseCv.wait(lock, [this, oldState]
+
+  if (timeout_ms) {
```

Comment:
Suggested comment:
```
// Since _phaseCv is a condition var., it should be prepared for spurious wakeups.
// So, waking it up on a timeout is harmless.
```